src/streamlink/plugins/nicolive.py

```diff
@@ -0,0 +1,330 @@
+import json
+import logging
+import re
+import websocket
+import threading
+import time
+
+from streamlink.plugin import Plugin, PluginArguments, PluginArgument
+from streamlink.plugin.api import useragents
+from streamlink.stream import HLSStream
+from streamlink.compat import urlparse
+
+
+_log = logging.getLogger(__name__)
+
+_url_re = re.compile(
+    r"^http(|s)://(?P<domain>live[0-9]*\.nicovideo\.jp)/watch/lv[0-9]*")
+
+_login_url = "https://account.nicovideo.jp/login/redirector?" \
+             "show_button_twitter=1&" \
```

Comment:
you could use this as **params** with `self.session.http.get(... params={'...': ...})` instead of making a long url with `\`